pkg/server/backend_manager.go

```diff
@@ -259,10 +278,10 @@ func (s *DefaultBackendStorage) RemoveBackend(identifier string, idType header.I
 	}
 	var found bool
 	for i, c := range backends {
-		if c.conn == conn {
+		if c == backend {
```

Comment:
Any risk that multiple `backend` structs will be constructed for a single connection? If so, it might be safer to check `c.conn == backend.conn`

Reply:
Done.

Currently that risk does not exist. But (after changing Backend from interface to type in latest snapshot) I converted the comparison also to be consistent with before.